core/src/apps/stellar/operations/layout.py

```diff
@@ -86,6 +92,16 @@
 
 
 async def confirm_manage_offer_op(ctx, op: StellarManageOfferOp):
+    text = confirm_manage_offer_common(op)
+    await _confirm_offer(ctx, text, op)
+
+
+async def confirm_manage_buy_offer_op(ctx, op: StellarManageBuyOfferOp):
+    text = confirm_manage_offer_common(op)
+    await _confirm_buy_offer(ctx, text, op)
+
+
+def confirm_manage_offer_common(op: StellarManageOfferUnion):
     if op.offer_id == 0:
```

Comment:
The test is passing in an operation where `offer_id` is `None`, which goes to the else-branch here, and then fails to format as number.

Given that `write_manage_offer_op` is serializing it as uint64 always, I think it's a problem with the test, and you should set the offer_id to 0 explicitly?

Reply:
I'm not sure about it − that's the original code here − but as far as I know, in transaction XDR `offer_id` is indeed set to 0 when creating new offers.

Reply:
what does your wallet do about `offer_id`?

Reply:
My wallet, nothing, but I believe `stellar-base` does set it to 0 when the parameter is not provided.

Reply:
makes sense. please modify the test case to set `offer_id` to 0 then